core/src/main/java/org/infinispan/atomic/impl/AtomicHashMap.java

```diff
@@ -54,35 +53,43 @@
    private volatile AtomicHashMapProxy<K, V> proxy;
    volatile boolean copied = false;
    volatile boolean removed = false;
+   volatile ProxyMode proxyMode = ProxyMode.NONE;
 
    /**
     * Construction only allowed through this factory method.  This factory is intended for use internally by the
     * CacheDelegate.  User code should use {@link AtomicMapLookup#getAtomicMap(Cache, Object)}.
     */
-   public static <K, V> AtomicHashMap<K, V> newInstance(Cache<Object, Object> cache, Object cacheKey) {
-      AtomicHashMap<K, V> value = new AtomicHashMap<K, V>();
+   @SuppressWarnings("unchecked")
+   public static <K, V> AtomicHashMap<K, V> newInstance(Cache<Object, Object> cache, Object cacheKey, ProxyMode proxyMode) {
+      AtomicHashMap<K, V> value = new AtomicHashMap<>(proxyMode);
       Object oldValue = cache.putIfAbsent(cacheKey, value);
       if (oldValue != null) value = (AtomicHashMap<K, V>) oldValue;
       return value;
    }
 
    public AtomicHashMap() {
-      this.delegate = new FastCopyHashMap<K, V>();
+      this(new FastCopyHashMap<>(), ProxyMode.NONE);
```

Comment:
I'm not sure the `NONE` option is necessary: I removed the enum value and replaced it here with `COARSE`, and all the tests passed.